EventListener/TagSubscriber.php

```diff
@@ -63,12 +86,16 @@ public function onKernelResponse(FilterResponseEvent $event)
     {
         $request = $event->getRequest();
         $response = $event->getResponse();
-
-        $tags = array();
+        if ($request->isMethodSafe()) {
```

Comment:
I prefer the exception, for it gets confusing for users if we let `addTags()` do invalidation too. The distinction makes sense: you have only one response during the request/response cycle that you want to set exactly the right set of tags on. While invalidating, however, you can just call `invalidateTags()` with some tag(s), and later in the cycle call it again with some other tag(s).